### ISSUE TITLE
Project Vision Documentation

### ISSUE THREAD

**Issue by Fraganya** [Fraganya/twilio-hook#1]
Develop project vision documentation to aid on-boarding interested devs!

**Comment by z-hymn** [comment]
Hi! I would like to help with this.

**Comment by Fraganya** [comment]
Hi @z-hymn ,Thank you for showing interest. Let me just organize some resources to help you provide better help .



> Hi! I would like to help with this.

**Comment by z-hymn** [comment]
Sounds great. I'll be on the lookout

**Comment by Fraganya** [comment]
Hello @z-hymn  , so i realized since you will write the tech document i just have to give some pointers as to what i think may be necessary to onboard someone, you will come up with the better content.

Going through the README , i hope you are able to grasp the whole concept of the idea? Let me know if there are any areas  you need clarifications on.

**Down to Business:**

What i envision is this:

- A developer should be able to setup this project , to some extent i feel that has been covered by the README but can be improved.
- A developer should be able to configure the app to respond to messages based on keywords or in a structured manner (USSD like)
- A developer should be able to setup a configuration of step options and associated methods
- The app should have a user interface/(Web or Mobile App) for the dev/user to monitor usage. basically some analytics.
- At some point (In the Future) the dev should be able to write programming logic and configure the steps in a convenient user interface.
- At some point (in the future) the app should be able to act as a hook for various chat bot APIs

So basically this is what i envision for the app.

**Comment by z-hymn** [comment]
So this will be my first project. I may need some more hands-on guidance as my background is in writing, and I am currently working on the technical side. This may take some more time as a maintainer. If that won't work for you and @laravel, I understand. 

The first issue I would like to point out is that in your ReadMe _Installation is misspelled._ 

I am taking more time to look at your ReadMe and seeing how it can be improved. I will get back to you again. Let me know if it's alright if a beginner can help you out. Thank you!

**Comment by Fraganya** [comment]
Great, feel free to reach out whenever you need any type of guidance. Take all the time you need. Our focus will be much on the technical implementation, you can come along with the nice documentation . 

Great notice on the README 👍🏾, i've just made the correction. Next time you can just edit and make a pull request ( Let me know if you need help on that)

Also note that i created the docs directory where we can be pushing all our documentation.

**Comment by z-hymn** [comment]
I went ahead and changed some of the sentence structures because I found the wording to be confusing. 

I’m thinking that most people who are using this webhook will be devs, so I took out any mention of devs. 

I also saw you wanted to emphasize certain words, and I changed the italicized words to convey what I think you were trying to emphasize. 

Let me know what you think. Since this is my first pr, I am big on feedback. Thanks!

**Comment by z-hymn** [comment]
Also, I was wondering if twilio-hook had a license, I put a heading at the very bottom for it.

**Comment by Fraganya** [comment]
Yea i have seen the changes and i think they are great , the sentences now sound better. 

On License , good point mentioning that out , i think we can use the MIT License.

**Comment by z-hymn** [comment]
Ok, will add update about MIT. Let me know about any other feedback. Thanks!
